crates/subspace-farmer/src/bin/subspace-farmer/commands/farm/dsn.rs

```diff
@@ -125,40 +127,47 @@ pub(super) async fn configure_dsn(
                                 Some(readers_and_pieces) => readers_and_pieces,
                                 None => {
                                     debug!(
-                                        ?req.piece_index_hash,
+                                        ?piece_index_hash,
                                         "Readers and pieces are not initialized yet"
                                     );
                                     return None;
                                 }
                             };
 
-                            readers_and_pieces.read_piece(&req.piece_index_hash)?
+                            readers_and_pieces
+                                .read_piece(&piece_index_hash)?
+                                .instrument(Span::current())
                         };
 
-                        let handle = piece_protocol_handle.clone();
-                        tokio::task::block_in_place(move || handle.block_on(read_piece_fut))
-                    }
-                };
+                        let piece = tokio::task::spawn_blocking(move || {
```

Comment:
Do we still need this complex code?

Reply:
Not really. The issue before was that above synchronous lock was held over `.await` so I moved it into blocking thread, then found that we don't actually need to clone so many things if we drop everything except the future (it was an owned value already) but somehow didn't notice that the future can be awaited in the upper level too just fine. Refactoring artifact, simplified, thanks!